internal/flags.go

```diff
@@ -107,59 +89,79 @@ func InitFlags(version string) (*ScriptBasedSoftwareUpdatableConfig, *logger.Log
 	if err := applyConfigurationFile(*configFile); err != nil {
 		return nil, nil, err
 	}
-
-	applyFlags(flg)
+	applyFlags(flgConfig)
 	return suConfig, logConfig, nil
 }
 
-func applyFlags(flg *cfg) {
+func initFlagsWithDefaultValues(config interface{}) {
+	valueOf := reflect.ValueOf(config).Elem()
+	typeOf := valueOf.Type()
+	for i := 0; i < typeOf.NumField(); i++ {
+		fieldType := typeOf.Field(i)
+		defaultValue := fieldType.Tag.Get("def")
+		description := fieldType.Tag.Get("descr")
+		fieldValue := valueOf.FieldByName(fieldType.Name)
+		pointer := fieldValue.Addr().Interface()
+		flagName := toFlagName(fieldType.Name)
+		switch val := fieldValue.Interface(); val.(type) {
+		case string:
+			flag.StringVar(pointer.(*string), flagName, defaultValue, description)
+		case int:
+			value, err := strconv.Atoi(defaultValue)
+			if err != nil {
+				log.Printf("error parsing integer argument %v with value %v", fieldType.Name, defaultValue)
```

Comment:
Is it the right way to handle this error i.e. there will be error log and the type default value will be set. Do we need to stop the excution?

Reply:
The same issue is open in file upload. Maybe they need to be resolved together.

Reply:
#15